ci_group/revolve2/ci_group/genotypes/cppnwin/modular_robot/v1/_body_develop.py

```diff
@@ -139,59 +129,26 @@ def __add_child(
     )
 
 
-def __add(a: tuple[int, int, int], b: tuple[int, int, int]) -> tuple[int, int, int]:
-    return (a[0] + b[0], a[1] + b[1], a[2] + b[2])
-
-
-def __timesscalar(a: tuple[int, int, int], scalar: int) -> tuple[int, int, int]:
-    return (a[0] * scalar, a[1] * scalar, a[2] * scalar)
-
-
-def __cross(a: tuple[int, int, int], b: tuple[int, int, int]) -> tuple[int, int, int]:
-    return (
-        a[1] * b[2] - a[2] * b[1],
-        a[2] * b[0] - a[0] * b[2],
-        a[0] * b[1] - a[1] * b[0],
-    )
-
-
-def __dot(a: tuple[int, int, int], b: tuple[int, int, int]) -> int:
-    return a[0] * b[0] + a[1] * b[1] + a[2] * b[2]
-
-
-def __rotate(
-    a: tuple[int, int, int], b: tuple[int, int, int], angle: int
-) -> tuple[int, int, int]:
+def __rotate(a: Vector3, b: Vector3, rotation: Quaternion) -> Vector3:
     """
     Rotates vector a a given angle around b.
 
-    Angle from [0,1,2,3].
-    90 degrees each.
-
     :param a: Vector a.
     :param b: Vector b.
-    :param angle: The angle to rotate.
+    :param rotation: The rotation Quaternion.
     :returns: A copy of a, rotated.
     """
-    cosangle: int
-    sinangle: int
-    if angle == 0:
-        cosangle = 1
-        sinangle = 0
-    elif angle == 1:
-        cosangle = 0
-        sinangle = 1
-    elif angle == 2:
-        cosangle = -1
-        sinangle = 0
-    else:
-        cosangle = 0
-        sinangle = -1
-
-    return __add(
-        __add(
-            __timesscalar(a, cosangle),
-            __timesscalar(__cross(b, a), sinangle),
-        ),
-        __timesscalar(b, __dot(b, a) * (1 - cosangle)),
-    )
+    cosangle = np.cos(rotation.angle)
+    sinangle = np.sin(rotation.angle)
+    x, y, z = (a * cosangle + b.cross(a) * sinangle) + b * (b.dot(a) * (1 - cosangle))
+    return Vector3(np.array([x, y, z], dtype=np.int64))
+
+
+def __make_quaternion_from_index(index: int) -> Quaternion:
```

Comment:
since we still want to rotate modules around their center to allow for differently directed hinges as an example, we have to translate the cppns output to a quaternion that allows for future rotation.